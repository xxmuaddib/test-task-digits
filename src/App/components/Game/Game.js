import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import * as S from './Game.styles';
import { Input } from '../../../ui/Input';
import { checkAnswer, updateInput } from '../../../redux/actions/game';
import { DigitsStatuses } from './components/DigitsStatuses';
import { GameResult } from './components/GameResult';

export const Game = () => {
  const dispatch = useDispatch();
  const { game } = useSelector((state) => state);

  const onCodeChange = (e) => {
    const code = e.target.value;

    if (code.length === 4) dispatch(checkAnswer(code));
    else dispatch(updateInput(code));
  };

  return (
    <S.GameContainer>
      <S.GameTitle className="animate__animated animate__slideInDown">
        Can you guess the secrets code?
      </S.GameTitle>
      <Input
        id="secret-code-input"
        value={game.input}
        onChange={onCodeChange}
        maxLength={4}
        placeholder="Secret code"
        disabled={!game.isGameStarted}
      />
      {game.secretCode && (
        <DigitsStatuses input={game.input} secretCode={game.secretCode} />
      )}
      {game.result !== null && (
        <GameResult result={game.result} correctAnswer={game.secretCode} />
      )}
    </S.GameContainer>
  );
};
