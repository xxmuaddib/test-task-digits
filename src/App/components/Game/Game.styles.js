import styled from 'styled-components';

export const GameContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 20px;
  height: 150px;
  width: 350px;
`;

export const GameTitle = styled.div`
  font-size: 24px;
`;
