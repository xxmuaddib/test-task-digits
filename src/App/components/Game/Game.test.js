import React from 'react';
import { Game } from './Game';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);

describe('Game', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      game: {
        secretCode: [1, 2, 3, 4],
        result: null,
        input: '',
        isGameStarted: true,
      },
    });

    component = mount(
      <Provider store={store}>
        <Game />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component).toMatchSnapshot();
  });
});
