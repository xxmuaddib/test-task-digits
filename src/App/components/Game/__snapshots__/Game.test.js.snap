// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should render with given state from Redux store 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Game>
    <styled.div>
      <div
        className="sc-bczRLJ hhtlfC"
      >
        <styled.div
          className="animate__animated animate__slideInDown"
        >
          <div
            className="sc-gsnTZi ianBro animate__animated animate__slideInDown"
          >
            Can you guess the secrets code?
          </div>
        </styled.div>
        <Input
          disabled={false}
          id="secret-code-input"
          maxLength={4}
          onChange={[Function]}
          placeholder="Secret code"
          value=""
        >
          <styled.div>
            <div
              className="sc-dkzDqf gQxYYg"
            >
              <styled.input
                disabled={false}
                id="secret-code-input"
                maxLength={4}
                onChange={[Function]}
                placeholder="Secret code"
                value=""
              >
                <input
                  className="sc-hKMtZM gATWiU"
                  disabled={false}
                  id="secret-code-input"
                  maxLength={4}
                  onChange={[Function]}
                  placeholder="Secret code"
                  value=""
                />
              </styled.input>
            </div>
          </styled.div>
        </Input>
        <DigitsStatuses
          input=""
          secretCode={
            Array [
              1,
              2,
              3,
              4,
            ]
          }
        >
          <styled.div>
            <div
              className="sc-eCYdqJ iyiAJB"
            />
          </styled.div>
        </DigitsStatuses>
      </div>
    </styled.div>
  </Game>
</Provider>
`;
